src/burn/engine/registration.cpp

```diff
@@ -1011,7 +1034,76 @@ extern "C" HRESULT RegistrationSaveState(
     }
     ExitOnFailure(hr, "Failed to write state to file: %ls", pRegistration->sczStateFile);
 
+    ::InitializeCriticalSection(&variables.csAccess);
+
+    hr = VariableDeserialize(&variables, TRUE, pbBuffer, cbBuffer, &iBuffer_Unused);
+    ExitOnFailure(hr, "Failed to read variables.");
+
+    // build variable registry key path
+    hr = StrAllocFormatted(&sczVariableKey, L"%s\\%s", pRegistration->sczRegistrationKey, REGISTRY_BUNDLE_VARIABLE_KEY);
+    ExitOnFailure(hr, "Failed to build variable registry key path.");
+
+    // open registration variable key
+    hr = RegCreate(pRegistration->hkRoot, sczVariableKey, KEY_WRITE | KEY_QUERY_VALUE, &hkRegistration);
+    ExitOnFailure(hr, "Failed to create registration variable key.");
+
+    hr = RegQueryInfoKey(hkRegistration, 0, 0, 0, 0, 0, 0, &dwNumberOfExistingValues, 0, 0, 0, 0);
+    ExitOnFailure(hr, "Failed to query registration variable count.");
+
+    for (DWORD i = dwNumberOfExistingValues; i >= 0; --i)
+    {
+        hr = RegValueEnum(hkRegistration, i, &sczValueName, &dwType);
+
+        if (E_NOMOREITEMS == hr)
+        {
+            hr = S_OK;
+            break;
+        }
+
+        ExitOnFailure(hr, "Failed to enumerate value %u", i);
+
+        hr = RegDeleteValue(hkRegistration, sczValueName);
+        ExitOnFailure(hr, "Failed to delete registration variable value.");
+    }
```

Comment:
I think you originally added this because we were going to get rid of the state file and enumerate the values instead. Can we remove this code now? The bundle should be writing the exact same variables every time so the values should always be overwritten. We're not clearing the values for any of the other registration keys.